recipes/antlr4-cppruntime/all/conanfile.py

```diff
@@ -23,133 +28,126 @@ class Antlr4CppRuntimeConan(ConanFile):
     short_paths = True
 
-    compiler_required_cpp17 = {
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
             "Visual Studio": "16",
+            "msvc": "1923",
             "gcc": "7",
             "clang": "5",
             "apple-clang": "9.1"
-    }
-
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+        }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        # We need this wrapper to build the library on Windows properly.
+        # We cannot use `cmake.configure(build_script_folder="runtime/Cpp")`,
+        # otherwise it is too late to set the option `WITH_STATIC_CRT`,
+        # and as a result the runtime library (< 4.11) is always built with static MSVC runtime.
+        # @see The function `def build(self)`.
+        copy(self, "CMakeLists.txt", src=self.recipe_folder, dst=os.path.join(self.export_sources_folder, "src"))
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("utfcpp/3.2.1")
-        if self.settings.os in ("FreeBSD", "Linux"):
-            self.requires("libuuid/1.0.3")
+        # As of 4.11, antlr4-cppruntime no longer requires libuuid.
+        # Reference: [C++] Remove libuuid dependency (https://github.com/antlr/antlr4/pull/3787)
+        # Note that the above PR points that libuuid can be removed from 4.9.3, 4.10 and 4.10.1 as well.
+        # We have backport the fix to these old versions.
 
     def validate(self):
-        if str(self.settings.arch).startswith("arm"):
-            raise ConanInvalidConfiguration("arm architectures are not supported")
-            # Need to deal with missing libuuid on Arm.
-            # So far ANTLR delivers macOS binary package.
+        compiler = self.info.settings.compiler
+        compiler_version = Version(compiler.version)
+        antlr_version = Version(self.version)
 
-        compiler = self.settings.compiler
-        compiler_version = tools.Version(self.settings.compiler.version)
-        antlr_version = tools.Version(self.version)
-
-        if compiler == "Visual Studio" and compiler_version < "16":
+        if (compiler == "Visual Studio" and compiler_version < "16") or \
+           (compiler == "msvc" and compiler_version < "1920"):
```

Comment:
It could be very simplified by using `check_min_vs_("1920")` 
https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#check-min-vs